src/api/Nhs.Appointments.Core/UserService.cs

```diff
@@ -19,10 +23,20 @@ public Task<string> GetApiUserSigningKey(string clientId)
     {
         return store.GetApiUserSigningKey(clientId);
     }
-    
-    public Task SaveUserAsync(string userId, string scope, IEnumerable<RoleAssignment> roleAssignments)
+
+    public async Task UpdateUserRoleAssignmentsAsync(string userId, string scope, IEnumerable<RoleAssignment> roleAssignments)
+    {
+        var oldRoles = await store.UpdateUserRoleAssignments(userId, scope, roleAssignments);
+
+        var rolesRemoved = oldRoles.Where(old => !roleAssignments.Any(r => r.Role == old.Role));
+        var rolesAdded = roleAssignments.Where(newRole => !oldRoles.Any(r => r.Role == newRole.Role));
+
+        await bus.Send(new UserRolesChanged { User = userId, Added = rolesAdded.Select(r => r.Role).ToArray(), Removed = rolesRemoved.Select(r => r.Role).ToArray()});
+    }
+
+    public async Task SaveUserAsync(string userId, string scope, IEnumerable<RoleAssignment> roleAssignments)
```

Comment:
Is this method now redundant? Also there is no need for this method to be marked as async  (which could be academic if it is being retired)